pkg/asset/installconfig/azure/client.go

```diff
@@ -286,11 +286,15 @@ func (c *Client) GetVMCapabilities(ctx context.Context, instanceType, region str
 	if typeMeta == nil {
 		return nil, fmt.Errorf("not found in region %s", region)
 	}
-
+	if typeMeta.Family == nil {
+		return nil, fmt.Errorf("error getting resource family")
+	}
 	capabilities := make(map[string]string)
 	for _, capability := range *typeMeta.Capabilities {
 		capabilities[to.String(capability.Name)] = to.String(capability.Value)
 	}
+	capabilities["Family"] = to.String(typeMeta.Family)
```

Comment:
Should we worry that Azure could, in theory, add a capability called `"Family"` and we'd be overriding it here?

Reply:
I guess that won't be a problem unless we need to eventually validate it, by which point we can rename this one.
/lgtm

Reply:
I've removed this and reworked it.